kong-ngx-build

```diff
@@ -265,8 +269,8 @@ main() {
       if [ ! -f Makefile ]; then
         OPENRESTY_OPTS=(
           "--prefix=$OPENRESTY_INSTALL"
-          "--with-cc-opt='-I$OPENSSL_INSTALL/include'"
-          "--with-ld-opt='-L$OPENSSL_INSTALL/lib -Wl,-rpath,$OPENSSL_INSTALL/lib'"
+          "--with-cc-opt='-I$OPENSSL_DESTDIR$OPENSSL_INSTALL/include'"
+          "--with-ld-opt='-L$OPENSSL_DESTDIR$OPENSSL_INSTALL/lib -Wl,-rpath,$OPENRESTY_RPATH'"
```

Comment:
By specifying `-rpath` here as `$OPENRESTY_RPATH`, it's expected that the resulting Nginx binary's RPATH contains $OPENRESTY_PATH? Does this output look correct:
```
$ objdump -a -x /tmp/build/usr/local/openresty/nginx/sbin/nginx | grep RPATH                                                 
  RPATH                /tmp/build/usr/local/openresty/luajit/lib:/tmp/build/usr/local/openssl/lib
```
Used the example command in the PR description, where `OPENRESTY_RPATH=/usr/local/kong/lib`.

Reply:
Yes, I believe this to be correct.

I'm partially relying on the available prior art
- https://github.com/Kong/kong-distributions/blob/master/kong-images/base-build.sh#L246
- https://github.com/Kong/kong-build-tools/blob/master/Dockerfile.kong#L69

Reply:
I double check the "final" result afer building / packaging / installing is:
```
objdump -a -x /usr/local/openresty/nginx/sbin/nginx | grep RPATH
  RPATH                /usr/local/openresty/luajit/lib:/usr/local/kong/lib
```

so I feel like this is correct(?)